### PR TITLE
doc,bgpv1: Add some failure scenarios

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
Add following new failure scenarios:

1. Peering Link Down
2. Cilium Operator Down
3. Service Losing All Backends

```release-note
doc,bgpv1: Add some failure scenarios
```

### CODE REVIEW

File: Documentation/network/bgp-control-plane-operation.rst [hunk 1]
```diff
@@ -67,3 +67,61 @@ all ongoing connections with the endpoints on the failed node, and connections
...
+and shuts down the TCP session right away. If a link not directly connected to
+the Node goes down, the BGP session will be lost after the hold timer expires,
+which is set to 90 seconds by default.
+
+Mitigation
+~~~~~~~~~~
+
+To make link detection failure fast, you can adjust ``holdTimeSeconds`` and
+``keepAliveTimeSeconds`` in the BGP configuration to the shorter value.
+However, the minimal possible value is ``holdTimeSeconds=3`` and
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
However, the minimal possible values are ``holdTimeSeconds=3`` and
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 2]
```diff
@@ -67,3 +67,61 @@ all ongoing connections with the endpoints on the failed node, and connections
...
+``keepAliveTimeSeconds=1``. The general approach to make it much faster is to
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
``keepAliveTimeSeconds=1``. The general approach to make failure detection faster is to
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 3]
```diff
@@ -67,3 +67,61 @@ all ongoing connections with the endpoints on the failed node, and connections
...
+use BFD (Bidirectional Forwarding Detection), but currently, Cilium does not
+support it.
+
+Cilium Operator Down
+--------------------
+
+If the Cilium Operator goes down, PodCIDR allocation by IPAM, and LoadBalancer
+IP allocation by LB-IPAM are stopped. Therefore, the new advertisement and
+withdraw of the PodCIDR and Service VIP routes will be stopped as well.
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
IP allocation by LB-IPAM are stopped. Therefore, the advertisement of new
and withdrawal of old PodCIDR and Service VIP routes will be stopped as well.
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 4]
```diff
@@ -67,3 +67,61 @@ all ongoing connections with the endpoints on the failed node, and connections
...
+
+Mitigation
+~~~~~~~~~~
+
+There's no direct mitigation in terms of the BGP. Kubernetes will eventually
+restart the Cilium Operator, and the routes will be advertised and withdrawn as
+usual.
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane-operation.rst]: Maybe we  should mention here that running more than 1 operator replicas may help to mitigate this?

File: Documentation/network/bgp-control-plane-operation.rst [hunk 5]
```diff
@@ -67,3 +67,61 @@ all ongoing connections with the endpoints on the failed node, and connections
...
+
+Service Losing All Backends
+---------------------------
+
+Service backends are all gone due to the outage or configuration mistake. BGP
+Control Plane behaves differently depending on the Service's
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
If all service backends are gone due to an outage or a configuration mistake, BGP
Control Plane behaves differently depending on the Service's
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 6]
```diff
@@ -67,3 +67,61 @@ all ongoing connections with the endpoints on the failed node, and connections
...
+``externalTrafficPolicy``. When the ``externalTrafficPolicy`` is set to
+``Cluster``, the Service's VIP kept advertised from all nodes selected by the
```
Comment (rastislavs) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
``Cluster``, the Service's VIP remains advertised from all nodes selected by the
```

**Review by rastislavs — CHANGES_REQUESTED** [review]
Nice write-up, just some minor corrections.

**Review by ghost — APPROVED** [review]
Docs structure looks good.

Can this change be backported to previous minor versions?

**Review by rastislavs — APPROVED** [review]
LGTM, thanks!

**Pull request merged**